### DISCUSSION TITLE
Adding $flow.fileName of upladed image to custom TOOL

### DISCUSSION THREAD

**Discussion by rec-it-yna** [FlowiseAI/Flowise#2349] [Ideas]
Dear,
i need to get the the imageURL of upladed image in the chat for example https://flowiseURL/api/v1/get-upload-file?chatflowId=<chatflowId>&chatId=<chatId>&fileName=<fileName> as a default variable in Function in custom tool. it has already only this $flow.sessionId 
$flow.chatId
$flow.chatflowId
$flow.input => this only return the text input of user
but i want to get the uploaded image URL ; my issue only is i can't fin the way to get the fileName of the upladed image. i need it for a custom function 
i know it's possible, here a screenshot of where it should modify in flowise code :
![1](https://github.com/FlowiseAI/Flowise/assets/153080985/c15d8536-c6a8-4eed-b45d-c99e1a77e134)
![2](https://github.com/FlowiseAI/Flowise/assets/153080985/e4af91af-908b-42e0-9a75-c1a5da58ad09)
![3](https://github.com/FlowiseAI/Flowise/assets/153080985/fe530f52-a51a-42c9-95a2-d484b88fb886)
![4](https://github.com/FlowiseAI/Flowise/assets/153080985/cab92c61-8a94-4c96-b3f1-ecaeeaae55c7)

it's realy an urgent need , and i really appreciate your support
thank you so much

**Comment by automaton82** [comment]
Why not make the changes and issue a PR?

**Reply by rec-it-yna** [reply]
thank you for your reply, i tried to set some change, but it taked something else wrong ;
